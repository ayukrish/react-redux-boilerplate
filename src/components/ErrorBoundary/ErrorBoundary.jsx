import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  children: PropTypes.node,
};

const defaultProps = {
  children: null,
};


class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: null };
  }

  componentDidCatch(error) {
    // Catch errors in any components below and re-render with error message
    this.setState({
      hasError: error,
    });
    // You can also log error messages to an error reporting service here
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return <h1>Something went wrong.</h1>;
    }

    return this.props.children;
  }
}

ErrorBoundary.propTypes = propTypes;
ErrorBoundary.defaultProps = defaultProps;

export default ErrorBoundary;
