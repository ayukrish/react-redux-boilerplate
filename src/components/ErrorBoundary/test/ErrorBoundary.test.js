import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ErrorBoundary from '../ErrorBoundary';

const Something = () => null;

describe('<ErrorBoundary> window', () => {
  it('should match the snapshot', () => {
    const tree = shallow(<ErrorBoundary>Test</ErrorBoundary>);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should display an ErrorMessage if wrapped component throws', () => {
    const tree = mount(
      <ErrorBoundary>
        <Something />
      </ErrorBoundary>,
    );

    const error = new Error('test');

    tree.find(Something).simulateError(error);
    tree.setState({ hasError: true });
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
