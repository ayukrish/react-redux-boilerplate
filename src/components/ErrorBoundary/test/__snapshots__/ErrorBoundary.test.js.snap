// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary> window should display an ErrorMessage if wrapped component throws 1`] = `
<ErrorBoundary>
  <h1>
    Something went wrong.
  </h1>
</ErrorBoundary>
`;

exports[`<ErrorBoundary> window should match the snapshot 1`] = `"Test"`;
